### PR TITLE
Externalize known server combinations

### LINKED ISSUE #93: Externalise the configuration of WellKnownFeaturePacks
All enum values in `WellKnownFeaturePacks` should be derived from an external config file.

This will allow to make sure that Prospero either deal with EAP or WildFly but not both. 

One way to externalise these values is with a properties file that list all required settings, with something like:

```
# list the keys of all known feature pack
prospero.feature-packs= x, y, z
# each feature pack defines its own properties
prospero.feature-pack.x.location=...
prospero.feature-pack.x.channel=...
# optional
prospero.feature-pack.x.pakages=...
# optional
prospero.feature-pack.x.repositories=...
```

### PR DESCRIPTION
**Pull request by spyrkob**
Fixes #93 

@TomasHofman could you take a look when you've got a moment

### CODE REVIEW

File: prospero-common/src/main/java/org/wildfly/prospero/api/WellKnownRepositories.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
  */
 public enum WellKnownRepositories implements Supplier<RemoteRepository> {
 
-    MRRC(new RemoteRepository.Builder("mrrc", "default", "https://maven.repository.redhat.com/ga/")
+    MRRC(new RemoteRepository.Builder("mrrc", "default", "https://maven.repository.redhat.com/earlyaccess/all/")
```
Comment (jmesnil) [inline prospero-common/src/main/java/org/wildfly/prospero/api/WellKnownRepositories.java:12]: Now that the feature pack location are externalised, why do you still need to know these remote repositories?

*If* they stay, I think you should distinguish between `MRRC` (or `MRRC_GA`) and `MRRC_EA`

File: prospero-common/src/main/resources/prospero-known-combinations.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (jmesnil) [inline prospero-common/src/main/resources/prospero-known-combinations.yaml:1]: -1, prospero is a community project and is not the place to provide product specific configuration

File: prospero-common/src/test/resources/prospero-known-combinations.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
+- name: "no-channel"
+  location: "wildfly@maven(org.jboss.universe:community-universe):current"
+  packages: []
```
Comment (jmesnil) [inline prospero-common/src/test/resources/prospero-known-combinations.yaml:4]: can the `packages` field be optional instead?

**Review by jmesnil — CHANGES_REQUESTED** [review]

File: prospero-common/src/main/java/org/wildfly/prospero/api/WellKnownRepositories.java (hunk 1 above)
Comment (spyrkob) [reply prospero-common/src/main/java/org/wildfly/prospero/api/WellKnownRepositories.java:12]: Mostly they're useful in tests, I should just move them to test tree though

File: prospero-common/src/main/resources/prospero-known-combinations.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+- name: "wildfly"
```
Comment (TomasHofman) [inline prospero-common/src/main/resources/prospero-known-combinations.yaml:2]: IIUC this is still not "provisionable" without providing some channel via `--channel` argument?

**Review by TomasHofman — APPROVED** [review]

File: prospero-common/src/main/resources/prospero-known-combinations.yaml (hunk 4 above)
Comment (spyrkob) [reply prospero-common/src/main/resources/prospero-known-combinations.yaml:2]: Yes the channel has to be provided separately

**Review by jmesnil — APPROVED** [review]

**Pull request merged**